From a367d9c0f33458c9cdf30a24d219e6584cd7f411 Mon Sep 17 00:00:00 2001
From: Rizky Benggolo <travarilo@gmail.com>
Date: Sat, 3 Jun 2023 22:55:00 +0000
Subject: [PATCH] Revert back clang version to clang-r475365b

---
 cc/config/global.go | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/cc/config/global.go b/cc/config/global.go
index 661f41eca..604c4b1cb 100644
--- a/cc/config/global.go
+++ b/cc/config/global.go
@@ -73,7 +73,7 @@ var (
 		"-Werror=int-conversion",
 
 		// Enable the new pass manager.
-		//"-fexperimental-new-pass-manager",
+		"-fexperimental-new-pass-manager",
 
 		// Disable overly aggressive warning for macros defined with a leading underscore
 		// This happens in AndroidConfig.h, which is included nearly everywhere.
@@ -186,10 +186,6 @@ var (
 	}
 
 	noOverrideGlobalCflags = []string{
-		// Workaround for boot loop caused by stack protector.
-		// http://b/267839238
-		"-mllvm -disable-check-noreturn-call",
-
 		"-Werror=bool-operation",
 		"-Werror=implicit-int-float-conversion",
 		"-Werror=int-in-bool-context",
@@ -252,9 +248,6 @@ var (
 		"-Wno-error=single-bit-bitfield-constant-conversion",
 		"-Wno-error=unused-private-field",
 		"-Wno-unused-command-line-argument",
-		"-Wno-error=thread-safety-analysis",
-		"-Wno-error=unguarded-availability",
-		"-Wno-error=logical-op-parentheses",
 	}
 
 	noOverrideExternalGlobalCflags = []string{
@@ -263,7 +256,6 @@ var (
 		"-Wno-unused-but-set-parameter",
 		// http://b/215753485
 		"-Wno-bitwise-instead-of-logical",
-		"-Wno-gnu-offsetof-extensions",
 	}
 
 	// Extra cflags for external third-party projects to disable warnings that
@@ -313,8 +305,8 @@ var (
 
 	// prebuilts/clang default settings.
 	ClangDefaultBase         = "prebuilts/clang/host"
-	ClangDefaultVersion      = "clang-r487747c"
-	ClangDefaultShortVersion = "17.0.2"
+	ClangDefaultVersion      = "clang-r475365b"
+	ClangDefaultShortVersion = "16.0.2"
 
 	// Directories with warnings from Android.bp files.
 	WarningAllowedProjects = []string{
-- 
2.34.1

